web/developer.actyx.com/docs/conceptual/performance-and-limits.mdx

```diff
@@ -122,15 +100,25 @@ We are working on removing this bottleneck in a future release.
 ### Pathological time travel performance when reconnecting a device that was offline for a long time
 
 When reconnecting a device that was disconnected for a long time period, this device will need to ingest all new events from all other devices in the swarm to become up-to-date.
-If a Fish is running during this process of catching up with the swarm, then that Fish will experience a lot more time travel than it would if it were woken up after Actyx is up-to-date.
 Actyx optimizes event delivery by doing it in chunks, nevertheless in pathological cases the amount of time travel is not linear in the number of such chunks but quadratic.
 
-As an example: consider 10 devices to catch up with, each contributing 10 chunks of events for your Fish.
-The optimal scenario would be to first catch up and then insert all 100 chunks into the event history, then perform one time travel.
-The worst-case scenario would be that every chunk causes a time travel, with each subsequent time travel being more costly than the previous (due to added events).
-Thereby, the Fish would process some events potentially a hundred times, more likely ten times (since chunks are ordered per device they come from).
+Event subscription via `subscribeMonotonic` and [machine-runner](https://www.npmjs.com/package/@actyx/machine-runner) may experience a lot more time travel than it would if it were woken up after Actyx is up-to-date.
+
+As an example: consider 10 devices to catch up with, each contributing 10 chunks of events for your application.
+At best, all 100 chunks are inserted into the event history and then one time travel is performed.
+At worst, every chunk causes a time travel, with each subsequent time travel being more costly than the previous (due to added events).
+Thereby, the application would process some events potentially a hundred times, more likely ten times (since chunks are ordered per device they come from).
 
-To avoid this issue, we recommend using `Pond.waitForSwarmSync()` or `Actyx.waitForSync()` to delay app startup until Actyx is up-to-date with the rest of the swarm.
+To avoid this issue, `Actyx.waitForSync()` can be used to delay app startup until Actyx is up-to-date with the rest of the swarm.
+
+### Startup Time in Machine-runner
+
+[machine-runner](https://www.npmjs.com/package/@actyx/machine-runner) works by subscribing to events of certain tags and applying all events that are found.
+At start up, a lot of matching events may have already been stored, which then makes the machine-runner take time, proportional to the number of events, until the first state is available.
+
+Our recommendation is to structure your machine-runner protocols such that each requires less than 1,000 events.
+This means that machine-runner are best used to model short-lived process workflows (which rarely have more than 1,000 steps).
```

Comment:
please turn “short-lived process workflows” into a link towards the best-practices page